lib/src/pubspec.dart

```diff
@@ -354,6 +355,57 @@ class Pubspec {
 
   Map<String, String> _executables;
 
+  /// The settings for which web compiler to use in which mode.
+  ///
+  /// It is a map of strings to string. Each key is the name of a mode, and
+  /// the value is the name of the web compiler to use in that mode.
+  ///
+  /// Valid compiler values are all of [Compiler.names].
+  Map<String, String> get webCompiler {
+    if (_webCompiler != null) return _webCompiler;
+
+    _webCompiler = {};
+    var webYaml = fields.nodes['web'];
+    if (webYaml == null) return _webCompiler;
+
+    if (webYaml is! Map) {
+      _error('"web" field must be a map.', webYaml.span);
+    }
+
+    var compilerYaml = (webYaml as YamlMap)['compiler'];
+    if (compilerYaml == null) return _webCompiler;
+
+    if (compilerYaml is! Map) {
+      _error('"compiler" field must be a map.',
+          (webYaml as YamlMap).nodes['compiler'].span);
+    }
+
+    compilerYaml.nodes.forEach((key, value) {
+      if (key.value is! String) {
+        _error('"compiler" keys must be strings.', key.span);
+      }
+
+      final keyPattern = new RegExp(r"^[a-zA-Z0-9_-]+$");
+      if (!keyPattern.hasMatch(key.value)) {
+        _error(
+            '"compiler" keys may only contain letters, '
+            'numbers, hyphens and underscores.',
+            key.span);
+      }
```

Comment:
Why this restriction? As far as I remember, there aren't any such restrictions on Barback mode names.